### PR TITLE
Re-align CI and test scenarios with Ember Defaults

### PR DESCRIPTION
**Pull request by jrjohnson**
- Removed old scenarios
- Update ember-try config to match new blueprints
- Extracted Node Test into their own step
- Bumped minimum node version to 12 (oldest current LTS)

WIP:
- [x] Should ember-auto-import be updated to v2 or added to try scenarios
- [x] Qunit v4 doesn't support the setup [method required](https://github.com/emberjs/ember-qunit/blob/master/docs/migration.md#qunit-dom) by v5, drop this?

### CODE REVIEW

**Comment by ro0gr** [comment]
> Should ember-auto-import be updated to v2 or added to try scenarios

ya, we'd need to do it anyways to support embroider. Let's upgrade auto-import.

> Qunit v4 doesn't support the setup method required by v5, drop this?

since it seems innevitable to upgrade the auto-import(which is a breaking change, as far as I understand), I think we may have troubles making the ember-qunit@4 to work with auto-import@2. After thinking about it more, I think I was wrong with my intention to support ember-qunit@4. Let's drop it. For those who may want to use it still, there is a v1 release.

**Comment by jrjohnson** [comment]
I'm stuck on jQuery (not a huge surprise). In Ember v4 the `jquery-integration` optional feature doesn't work anymore. Even if I install `@ember/jquery` though and turn off this flag I get `Ember is not defined` errors which I think come from https://github.com/ro0gr/ember-cli-page-object/blob/upgrade/ember%403.28/addon-test-support/adapters/acceptance.js#L3 (and a few other places with `const { $ } = Ember;`. I'm not sure familiar enough with what these different parts of the code do to know what's critical here.

Is it possible that in `v2` these sort of legacy jQuery handlers can go away or rely on the `-jquery` that gets bundled by this addon? Or is it impossible to support Ember v4 without first dropping some things? Or am I missing what's going on and there is a fix?

**Comment by ro0gr** [comment]
@jrjohnson I'd like to simplify jquery stuff a bit in scope of v2(https://github.com/san650/ember-cli-page-object/pull/556#discussion_r780681443). I believe the best path forward for it is to remove `adapters/{acceptance,integration,native-*}` adapters and switch to consume the `jquery` dependency for some remaining places in the codebase. Feel free to remove those old adapters. I believe this should allow us to import jquery directly w/o a need for shims or some other magic:
```
import $ from 'jquery';
```

In general I'd like to get rid of jquery completely of course, but it should probably be done after the v2 is released.

**Comment by jrjohnson** [comment]
Just an update on this. I've mostly converted to `import $ from 'jquery';`, however that is getting transpired somewhere to `Ember.$`. I'm trying to track down where/why that is happening.

**Comment by ro0gr** [comment]
This is strange. It should have been ignored due to https://github.com/jrjohnson/ember-cli-page-object/blob/align-scenarios-with-328/babel.config.js#L10

File: tests/dummy/config/optional-features.json [hunk 1]
```diff
@@ -1,3 +1,6 @@
 {
-  "jquery-integration": true
-}
+  "application-template-wrapper": false,
+  "default-async-observers": true,
+  "jquery-integration": false,
```
Comment (ro0gr) [inline tests/dummy/config/optional-features.json]: I suspect if revert this back to `true`, CI would work again, but ember@4, right? 
If such, I'd suggest to go ahead and disable ember@4 scenarios for now, and re-enable them in scope of the embroider PR.

Comment (jrjohnson) [reply tests/dummy/config/optional-features.json]: Ok, that does the trick. All set now.

**Pull request merged**